### PR TITLE
Normalize path input for jstree tree

### PR DESCRIPTION
**Pull request by MindTooth**
Attempt at fixing reported issue.

@tomrtk thoughts?  Related to reported security issue.

### CODE REVIEW

File: src/core/services.py [hunk 1]
```diff
@@ -655,7 +655,7 @@ def get_directory_listing(
     else:
         directory = path
 
-    normalized_path = pathlib.Path(directory)
+    normalized_path = pathlib.Path(os.path.normpath(directory))
```
Comment (tomrtk) [inline src/core/services.py:658]: if it do not resolve the issue, may need something like this as well, to catch `../../../xxx` type of paths.
```py
    normalized_path = pathlib.Path(os.path.normpath(directory))
    if not str(normalized_path).startswith(directory):
        raise Exception("fishy path")
```
and remove the else branch above.

**Review by tomrtk — APPROVED** [review]
lgtm, but not sure if it will fix the issue, I read it as more due to the fundamental design.

File: src/core/services.py (hunk 1 above)
Comment (MindTooth) [reply src/core/services.py:658]: Thanks for the suggestion.  I'll implement something like this if CodeQL does not get happy with this. 😄

**Comment by MindTooth** [comment]
Still needs fixing. 😭   Guess it's very sensitive about user input.

**Comment by tomrtk** [comment]
> Still needs fixing. 😭 Guess it's very sensitive about user input.

maybe something like this..
```py
    # can also be stored as a module constant
    video_root = config.get( 
            "CORE",
            "video_root_path",
            fallback=str(get_video_root_path()),
    )
    if path is None:
        directory = video_root
    else:
        directory = path

    normalized_path = pathlib.Path(os.path.normpath(directory))
    if not str(normalized_path).startswith(video_root):
        raise SomeException("fishy path")
```
then it should not allow any paths, not having `video_root` as parent.

**Pull request merged**